packages/adblocker-electron/content.ts

```diff
@@ -0,0 +1,180 @@
+/*!
+ * Copyright (c) 2017-2019 Cliqz GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+import {
+  IBackgroundCallback,
+  IMessageFromBackground,
+} from '@cliqz/adblocker-webextension-cosmetics';
+import { ipcRenderer, webFrame } from 'electron';
+
+import { extractFeaturesFromDOM } from '@cliqz/adblocker';
+
+const enableMutationObserver = true; // Not sure what to do with it.
+
+declare global {
+  interface Window {
+    MutationObserver?: typeof MutationObserver;
+  }
+}
+
+let ACTIVE: boolean = true;
+let MUTATION_OBSERVER: MutationObserver | null = null;
+
+
+function unload(): void {
+  if (MUTATION_OBSERVER !== null) {
+    MUTATION_OBSERVER.disconnect();
+    MUTATION_OBSERVER = null;
+  }
+}
+
+function handleResponseFromBackground({ active, scripts }: IMessageFromBackground): void {
+  if (active === false) {
+    ACTIVE = false;
+    unload();
+    return;
+  } else {
+    ACTIVE = true;
+  }
+
+  // Inject scripts
+  if (scripts) {
+    for (let i = 0; i < scripts.length; i += 1) {
+      setTimeout(() => webFrame.executeJavaScript(scripts[i]), 0);
+    }
+  }
+}
+
+function getCosmeticsFilters(data: IBackgroundCallback) {
+  const id =
```

Comment:
As suggested earlier, maybe we could get rid of `id` completely (check comment in `adblocker.ts`). It would also mean we could have only one message listener registered as soon as preload script is injected in a page and listening to channel `get-cosmetic-filters-response`.